src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,205 @@
+
+    public static void main(String[] args) {
+
+        String line;
+        Scanner in = new Scanner(System.in);
+
+        sayHello();
+        while (isRunning) {
+            line = in.nextLine();
+            try {
+                if (line.trim().equals("bye")) {
+                    exitProgram();
+                } else if(line.trim().equals("list")) {
+                    printList();
+                } else if (line.startsWith("done")) {
+                    makeDone(line);
+                } else if (line.startsWith("todo")) {
+                    addTodoTask(line);
+                } else if (line.startsWith("deadline")) {
+                    addDeadlineTask(line);
+                } else if (line.startsWith("event")) {
+                    addEventTask(line);
+                } else {
+                    throw new InvalidCommandException();
+                }
+            } catch (InvalidCommandException e) {
+                sayInvalid();
+            }
+        }
+        sayBye();
+    }
+
+    private static void sayHello() {
+        System.out.println(LOGO);
+        System.out.println(HORIZONTAL_LINE);
+        System.out.println("Hello! I'm Bobo!");
+        System.out.println("I'm a little blur, but I'd love to help!");
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    private static void sayBye() {
+        System.out.println(HORIZONTAL_LINE);
+        System.out.println("Ok. Bye bye!");
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    private static void sayInvalid() {
+        System.out.println(HORIZONTAL_LINE);
+        System.out.println("Sorry I don't understand that! Can you rephrase?");
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    private static void printList() {
+        System.out.println(HORIZONTAL_LINE);
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < numberOfTasks; i++) {
+            System.out.println((i + 1) + "." + tasks[i]);
+        }
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    private static void printDone(int taskNumber) {
+        System.out.println(HORIZONTAL_LINE);
+        System.out.println("Okie! Marked this as done:");
+        System.out.println((taskNumber + 1) + "." + tasks[taskNumber]);
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    private static void makeDone(String line) {
+        try {
+            handleDone(line);
+        } catch (NumberFormatException e) {
+            printMissingDoneIndexMsg();
+        } catch (NullPointerException e) {
+            printInvalidTaskIndexMsg();
+        }
+    }
+
+    private static void handleDone(String line) {
+        int taskNumber = Integer.parseInt(line.substring(END_OF_DONE_INDEX).trim()) - 1;
+        if (tasks[taskNumber].isDone()) {
+            System.out.println("This task is already done!");
+        } else {
+            tasks[taskNumber].markAsDone();
+            printDone(taskNumber);
+        }
+    }
```

Comment:
Consider creating a TaskManager class to handle manipulation of tasks e.g. deadline, todo etc. Also the class can contain the list of tasks instead of it being at the start of main